### PR TITLE
Added User data TextField validators

### PR DESCRIPTION
**Pull request by Bahaaib**
Track issue here: 
#54

### CODE REVIEW

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please, pay attention to this pull request

**Comment by 0crat** [comment]
Job [#55](https://github.com/ammaratef45/attendance/issues/55) is now in scope, role is `REV`

**Comment by 0crat** [comment]
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) everybody who has role `REV` is banned at [#55](https://github.com/ammaratef45/attendance/issues/55); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by ammaratef45** [comment]
@Bahaaib can you merge master in your branch?

File: attendance/lib/backend/user.dart [hunk 1]
```diff
@@ -29,7 +29,7 @@ class User {
   /// rename the user (change native name)
   void rename(String newName) {
     final String trimmed = newName.trim();
-    if (_isValidName(trimmed)) {
+    if (isValidName(trimmed)) {
```
Comment (ammaratef45) [inline attendance/lib/backend/user.dart]: @Bahaaib no need to make`isValidName` public, I guess you can just catch the exception and deal with it

File: attendance/lib/backend/user.dart [hunk 2]
```diff
@@ -39,22 +39,24 @@ class User {
   /// change user's phone number
   void changePhone(String newPhone) {
     final String trimmed = newPhone.trim();
-    if (_isValidNumber(trimmed)) {
+    if (isValidNumber(trimmed)) {
```
Comment (ammaratef45) [inline attendance/lib/backend/user.dart]: @Bahaaib The same as in `isValidName`

File: attendance/lib/ProfilePage/profile_page_view.dart [hunk 3]
```diff
@@ -195,16 +204,41 @@ class ProfilePageView extends ProfilePageViewModel {
...
+    if (User.instance().isValidName(name)) {
+      return null;
+    } else {
+      return 'Invalid name format';
+    }
+  }
+
+  String _phoneValidator(String number) {
+    if (User.instance().isValidNumber(number)) {
+      return null;
```
Comment (ammaratef45) [inline attendance/lib/ProfilePage/profile_page_view.dart]: @Bahaaib Avoid the usage of `null`

**Comment by ammaratef45** [comment]
@rultor merge

**Comment by ammaratef45** [comment]
@Bahaaib Thanks

**Comment by ammaratef45** [comment]
@Bahaaib I should fix tests with rultor as I did with circleci, I will do it

**Comment by 0crat** [comment]
Job `gh:ammaratef45/attendance#55` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#55](https://github.com/ammaratef45/attendance/issues/55) is now out of scope

**Pull request merged**